module-system/sov-state/src/storage.rs

```diff
@@ -62,52 +64,85 @@ impl StorageKey {
     }
 }
 
-// `Value` type for the `Storage`
-#[derive(Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+/// A serialized value suitable for storing. Internally uses an Arc<Vec<u8>> for cheap cloning.
+#[derive(
+    Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize, Serialize, Deserialize, Default,
+)]
 pub struct StorageValue {
     value: Arc<Vec<u8>>,
 }
 
+impl From<CacheValue> for StorageValue {
+    fn from(cache_value: CacheValue) -> Self {
+        Self {
+            value: cache_value.value,
+        }
+    }
+}
+
+impl From<Vec<u8>> for StorageValue {
+    fn from(value: Vec<u8>) -> Self {
+        Self {
+            value: Arc::new(value),
```

Comment:
Unrelated to the PR, but I don't see this specific instance being sent across threads. I think we fetch the owned value from the lock, and use it

Reply:
Let's open an issue for that